### PR TITLE
feat(gradle): Add basic Gradle Kotlin DSL support

### PR DESCRIPTION
**Pull request by ikesyo**
This partially implements #3054.

- `*.gradle.kts` ~~and `buildSrc/**/*`~~ files are now recognized
- Kotlin's named arguments are supported, such as: `implementation(group = "...", name = "...", version = "...")`

### CODE REVIEW

**Comment by rarkins** [comment]
Thanks @ikesyo ! Hoping that @corecanarias has some spare time to review this soon

**Comment by ikesyo** [comment]
Hi @corecanarias @rarkins, is there any chance to get this reviewed?

**Comment by rarkins** [comment]
@ikesyo thank you for this PR! I think @corecanarias is busy lately, but I'll try to take a look. Could you fix the conflicted file?

**Comment by ikesyo** [comment]
> Could you fix the conflicted file?

Yes of course!

**Comment by ikesyo** [comment]
@rarkins I've resolved the conflicts.

File: lib/config/definitions.js [hunk 1]
```diff
@@ -1624,7 +1624,11 @@ const options = [
     stage: 'package',
     type: 'object',
     default: {
-      fileMatch: ['\\.gradle$', '(^|/)gradle.properties$'],
+      fileMatch: [
+        '\\.gradle(\\.kts)?$',
+        '(^|/)gradle.properties$',
+        '(^|/)buildSrc/.*',
```
Comment (rarkins) [inline lib/config/definitions.js]: Why is this needed in the `fileMatch`?

Comment (ikesyo) [reply lib/config/definitions.js]: As @jcornaz said in #3054, it is becoming popular that defining dependencies/versions in a Kotlin source files under `buildSrc`. For example:

- https://handstandsam.com/2018/02/11/kotlin-buildsrc-for-better-gradle-dependency-management/
- https://proandroiddev.com/gradle-dependency-management-with-kotlin-94eed4df9a28

Even though I don't handle contents such as:

```kotlin
object Versions {
    val support_lib = "27.0.2"
    val retrofit = "2.3.0"
    val rxjava = "2.1.9"
}

object Libs {
    val support_annotations = "com.android.support:support-annotations:${Versions.support_lib}"
    val support_appcompat_v7 = "com.android.support:appcompat-v7:${Versions.support_lib}"
    val retrofit = "com.squareup.retrofit2:retrofit:${Versions.retrofit}"
    val retrofit_rxjava_adapter = "com.squareup.retrofit2:adapter-rxjava2:${Versions.retrofit}"
    val rxjava = "io.reactivex.rxjava2:rxjava:${Versions.rxjava}"
}
``` 

but this PR should work against more simple cases such as:

```kotlin
object Libs {
    private val supportLibVersion = "27.0.2"
    private val retrofitVersion = "2.3.0"
    private val rxjavaVersion = "2.1.9"

    val support_annotations = "com.android.support:support-annotations:${supportLibVersion}"
    val support_appcompat_v7 = "com.android.support:appcompat-v7:${supportLibVersion}"
    val retrofit = "com.squareup.retrofit2:retrofit:${retrofitVersion}"
    val retrofit_rxjava_adapter = "com.squareup.retrofit2:adapter-rxjava2:${retrofitVersion}"
    val rxjava = "io.reactivex.rxjava2:rxjava:${rxjavaVersion}"
}
```

Comment (ikesyo) [reply lib/config/definitions.js]: I reconsidered this. As you said in https://github.com/renovatebot/renovate/issues/3054#issuecomment-452850852, not adding `buildSrc` to the `fileMatch` and letting users do that by themselves will be a good (at least for now) as a first step.

Comment (ikesyo) [reply lib/config/definitions.js]: I've just omitted the entry.

Comment (ikesyo) [reply lib/config/definitions.js]: @rarkins I've just omitted the entry.

**Review by corecanarias — APPROVED** [review]
Sorry it took so long :/

**Comment by rarkins** [comment]
Thanks @ikesyo for the feature and @corecanarias for the review!
